table/block_based/filter_policy.cc

```diff
@@ -1311,6 +1294,42 @@ Status XXPH3FilterBitsBuilder::MaybePostVerify(const Slice& filter_content) {
 }
 }  // namespace
 
+Slice FilterBitsBuilder::Finish(std::unique_ptr<const char[]>* buf) {
+  Slice filter;
+  Status s = FinishV2(/*allocator*/ nullptr, &filter);
+  if (filter.size() > 0) {
+    assert(s.ok());
+    buf->reset(filter.data());
```

Comment:
(Just a thought) Compared with the previous API where we can directly pass in the `std::unique_ptr<const char[]>* buf` and initiate it right on allocating the data, things like `buf->reset(filter.data());` open up possibility for extra pointers to the underlying data pointed by `std::unique_ptr<const char[]>* buf`. For example, we can maliciously do `KDefaultMemoryAllocator.Deallocate(const_cast<char*>(filter.data()))`. 

I don't have a way to solve this given that we need the compatibility so this is just a thought (probably a useless one too :)